src/caioaao/kaocha_greenlight/test/var.clj

```diff
@@ -15,19 +15,22 @@
    :kaocha.result/pending (:pending rs 0)
    :kaocha.result/count   1})
 
-(defn report [options event]
-  (when (= :step-end (:type event))
-    (->> (:step event)
-         ::step/reports
-         (run! ctest/do-report)))
-  (ctest/do-report event))
+(defn report [event]
+  (let [{:keys [step type]}                     event
+        {::step/keys [outcome reports message]} step]
+    (when (= type :step-end)
+      (run! ctest/do-report reports))
+    (when (= outcome :error)
+      (ctest/report {:type    :error
```

Comment:
The failed step doesn't contain the actual exception, which is why I used `report` instead of `do-report`. I created an issue to track adding the actual exception to the event: https://github.com/amperity/greenlight/issues/50

Reply:
Created a PR to add this functionality: https://github.com/amperity/greenlight/pull/51